helix-lsp/src/lib.rs

```diff
@@ -264,6 +294,90 @@ impl Registry {
     }
 }
 
+#[derive(Debug)]
+pub enum ProgressStatus {
+    Created,
+    Started(lsp::WorkDoneProgress),
+}
+
+impl ProgressStatus {
+    pub fn progress(&self) -> Option<&lsp::WorkDoneProgress> {
+        match &self {
+            ProgressStatus::Created => None,
+            ProgressStatus::Started(progress) => Some(&progress),
+        }
+    }
+}
+
+#[derive(Debug)]
+/// Acts as a container for progress reported by language servers. Each server
+/// has a unique id assigned at creation through `Registry`. This id is then used
+/// to store the progress in this map.
+pub struct LspProgressMap(Vec<HashMap<lsp::ProgressToken, ProgressStatus>>);
```

Comment:
I think you want this to be a `HashMap<id, HashMap>` so that we're able to remove entries if a certain LSP shuts down (they don't do that yet but that could be a thing in the future).

Reply:
Done.